common/network-common/src/main/java/org/apache/spark/network/buffer/FileSegmentManagedBuffer.java

```diff
@@ -92,12 +91,27 @@ public ByteBuffer nioByteBuffer() throws IOException {
   }
 
   @Override
-  public InputStream createInputStream() throws IOException {
+  public InputStream createInputStream(boolean checksum) throws IOException {
```

Comment:
So, is this only for testing? because it's otherwise very expensive to compute, requiring two passes over the file.

If so, should it not be in some more package-private method only and not exposed?